File: Lab6_7_Sorting_algorithms/bubble_sort/bubble_sort.py
def bubble_sort(arr):
    n = len(arr)
    # Проходим по всем элементам массива
    for i in range(n - 1):
        for j in range(n - i - 1):
            # Сравниваем соседние элементы
            if arr[j] > arr[j + 1]:
                # Меняем местами, если они стоят в неправильном порядке
                arr[j], arr[j + 1] = arr[j + 1], arr[j]


# Функция для вывода массива
def print_array(arr):
    for num in arr:
        print(num, end=" ")
    print()


# Пример использования
if __name__ == "__main__":
    arr = [64, 34, 25, 12, 22, 11, 44]

    print("Исходный массив:")
    print_array(arr)

    bubble_sort(arr)

    print("Отсортированный массив:")
    print_array(arr)
# Начальный массив: [64, 34, 25, 12, 22, 11, 44]
# Массив после первой итерации: [34, 25, 12, 22, 11, 44, 64]

File: Lab6_7_Sorting_algorithms/bubble_sort/bubble_sort.java
import java.util.*;


public class Main {
    public static void bubbleSort(int[] arr) {
        int n = arr.length;
        // Проходим по всем элементам массива
        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                // Сравниваем соседние элементы
                if (arr[j] > arr[j + 1]) {
                    // Меняем местами, если они стоят в неправильном порядке
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
    }

    // Функция для вывода массива
    public static void printArray(int[] arr) {
        for (int num : arr) {
            System.out.print(num + " ");
        }
        System.out.println();
    }

    // Пример использования
    public static void main(String[] args) {
        int[] arr = {64, 34, 25, 12, 22, 11, 44};

        System.out.println("Исходный массив:");
        printArray(arr);

        bubbleSort(arr);

        System.out.println("Отсортированный массив:");
        printArray(arr);
    }
}
// Начальный массив: [64, 34, 25, 12, 22, 11, 44]
// Массив после первой итерации: [34, 25, 12, 22, 11, 44, 64]
// Итоговый массив: [11, 12, 22, 25, 34, 44, 64]
